### PR TITLE
chore: try to disable flipper for ci

### PR DESCRIPTION
**Pull request by gkartalis**
This PR resolves []

### Description

Read recently that we can disable flipper in CI in order to save some resources and make the iOS build faster when building the iOS app in circleCI.

here are the [docs](https://fbflipper.com/docs/getting-started/troubleshooting/react-native/#disable-flipper-on-ci-builds-ios) for this

Mostly trying out [this solution](https://github.com/facebook/flipper/issues/1326#issuecomment-788223247)

Happy to hear your thoughts on this.

## Results
Taking the logs from `--- Step: build_ios_app ---` to `Successfully exported and compressed dSYM file` of the `Deploy if Beta` step.
- before
https://app.circleci.com/pipelines/github/artsy/eigen/27991/workflows/364fc04b-7526-4b82-810a-2fcba217ad3b/jobs/83420

```
[17:04:59]: --- Step: build_ios_app ---
[17:18:01]: Successfully exported and compressed dSYM file
```

Build took **13m 02s**

- after
https://app.circleci.com/pipelines/github/artsy/eigen/28024/workflows/d566018f-fec6-4b43-9624-9665b96b66cc/jobs/83499

```
[08:54:15]: --- Step: build_ios_app ---
[09:03:33]: Successfully exported and compressed dSYM file
```

Build took **9m 18s**

### PR Checklist

- [ ] I tested my changes on **iOS** / **Android**.
- [ ] I added screenshots or videos to illustrate my changes.
- [ ] I added Tests and Stories for my changes.
- [ ] I added an [app state migration].
- [ ] I hid my changes behind a [feature flag].

### To the reviewers 👀

- [ ] I would like **at least one** of the reviewers to run this PR on the simulator or device.

<details><summary>Changelog updates</summary>

#### Cross-platform user-facing changes

-

#### iOS user-facing changes

-

#### Android user-facing changes

-

#### Dev changes

- remove flipper from iOS ci - gkartalis

</details>

Need help with something? Have a look at our [docs], or get in touch with us.

[app state migration]: ../blob/main/docs/adding_state_migrations.md
[feature flag]: ../blob/main/docs/developing_a_feature.md
[docs]: ../blob/main/docs/README.md

### CODE REVIEW

File: Podfile [hunk 1]
```diff
@@ -154,12 +154,18 @@ target 'Artsy' do
 end
 
 
-# Enables Flipper.
+# Enables Flipper but not for CI.
 # Note that if you have use_frameworks! enabled, Flipper will not work and
 # you should disable these next few lines.
-use_flipper!({ 'Flipper' => '0.87.0', 'Flipper-Folly' => '2.5.3',  'Flipper-RSocket' => '1.3.1' })
+if !ENV['CI']
+  use_flipper!({ 'Flipper' => '0.87.0', 'Flipper-Folly' => '2.5.3',  'Flipper-RSocket' => '1.3.1' })
```
Comment (pvinis) [inline Podfile:161]: probably we can remove the hardcoded versions now.. not for this PR though.

File: Artsy/App/ARAppDelegate.m [hunk 2]
```diff
@@ -46,8 +46,7 @@
 #import <Emission/AREmission.h>
 #import <Emission/ARNotificationsManager.h>
 
-#ifdef FB_SONARKIT_ENABLED
-#import <FlipperKit/FlipperClient.h>
+#if defined(FB_SONARKIT_ENABLED) && __has_include(<FlipperKit/FlipperClient.h>)
```
Comment (pvinis) [inline Artsy/App/ARAppDelegate.m:49]: nice. I prefer if defined to ifdef anyway, and it's a good use here 👌

**Review by pvinis — APPROVED** [review]
awesome! thank you for taking the time for this PR!

File: Artsy/App/ARAppDelegate.m (hunk 2 above)
Comment (pvinis) [inline Artsy/App/ARAppDelegate.m:49]:
```suggestion
#if defined(FB_SONARKIT_ENABLED) && __has_include(<FlipperKit/FlipperClient.h>)
#import <FlipperKit/FlipperClient.h>
```

**Comment by pvinis** [comment]
#squashongreen

**Comment by pvinis** [comment]
I added some results. Before the build took 13m 2s, after it took 9m 18s. That's a 30% decrease, and an actual diff of 4 minutes less.

I'm sure we can do more to remove time from this workflow, but I think that's a pretty good one!

**Comment by pvinis** [comment]
GH shows failing checks, but it's fine. the android one passed on the last one, and the ios one failed because the appstoreconnect doesn't have the next version ready. I will merge this. Thank you @gkartalis!

**Comment by gkartalis** [comment]
So excited for this! It will save us so much time and resource in the long 🏃🏻! Thanks for helping polishing and pushing this through the finish line 🔥🔥🔥🔥🔥

**Pull request merged**